### PR TITLE
Restore CPU offloading capability

### PR DESCRIPTION
**Pull request by aubreyli**
Restore CPU offloading capability.
Provided an example yaml for deepseek-V2-Lite-Chat model.
Patch tested by both default GPU case and GPU+CPU case.

### CODE REVIEW

**Comment by Azure-Tang** [comment]
Thank you for your contribution, this means KT restored cpu capacity.

Btw can you edit the target merge branch to Develop 0.2.3? Thank you!

**Comment by aubreyli** [comment]
Changed the base to develop-0.2.3

**Comment by aubreyli** [comment]
Uh, it should be clean now。

**Comment by Azure-Tang** [comment]
deepseek-V2-Lite-Chat use the same yaml for deepseek-V2, so there is no need to add a new yaml~

**Comment by aubreyli** [comment]
This new yaml shows a GPU+CPU mixed exmaple, about how to allocate different modules and different layers to GPU and CPU, it seems we don’t have such an example.

File: ktransformers/operators/models.py [hunk 1]
```diff
@@ -649,14 +649,10 @@ def forward(
...
-                # only use mask in forward windows or can't flash attn
-                causal_mask = self._update_causal_mask(
-                    attention_mask, inputs_embeds, cache_position, past_key_values, output_attentions
-                )
-            else:
-                causal_mask = None
+            # only use mask in forward windows or can't flash attn
+            causal_mask = self._update_causal_mask(
+                attention_mask, inputs_embeds, cache_position, past_key_values, output_attentions
+            )
```
Comment (Atream) [inline ktransformers/operators/models.py]: Using an attention mask will incur additional memory overhead that is quadratic with respect to the context length, which can significantly reduce the context length. Let's think about how to make a judgment here, and create the mask when using CPU for Attention.

Comment (aubreyli) [reply ktransformers/operators/models.py]: @Atream I refined the patch, please help to review again

**Comment by aubreyli** [comment]
@Azure-Tang I saw 0.2.3 has been released, so I changed the base to main, but please let me know if you still want me to rebase to develop-0.2.3
